pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -488,6 +491,7 @@ public void preloadSegment(ImmutableSegment segment) {
     try {
       doPreloadSegment((ImmutableSegmentImpl) segment);
       _trackedSegments.add(segment);
+      _updatedSegmentsSinceLastSnapshot.add(segment);
```

Comment:
preloadSegment() checked _enableSnapshot above already, so saved the if-check here.